### PR TITLE
test: add a test for existing many partition tables

### PR DESCRIPTION
**Pull request by komainu8**
This test for confirming a modification of https://github.com/pgroonga/pgroonga/commit/b54cd0ce34d13e2c0a9a473b6631c0efcaf40265.

The version 2.3.8 or less of PGroonga crashed in this test case.

### CODE REVIEW

File: sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    city varchar(3) NOT NULL,
+    n_infection varchar(205),
+    city_code varchar(5) NOT NULL,
+    remarks varchar(200)
+);
+ALTER TABLE ONLY infection_numbers_influenza ATTACH PARTITION infection_numbers_influenza_22_04 FOR VALUES IN ('22-04');
+
+INSERT INTO infection_numbers_influenza_20_04 VALUES ('Japan','Mie','Ise','134','20-04','age: 30, gender:male, overseas travel history: true');
+INSERT INTO infection_numbers_influenza_22_04 VALUES ('U.S.A','NY','NY','10221','22-04','age: 10, gender:male, overseas travel history: false');
+INSERT INTO infection_numbers_influenza_21_04 VALUES ('Germany','BER','BER','221','21-04','age: 50, gender:male, overseas travel history: false');
```
Comment (kou) [inline sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql:133]: Could you swap them for readability?

File: sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+CREATE TABLE infection_numbers_influenza_20_01 (
+    country varchar(16) NOT NULL,
+    prefecture varchar(4) NOT NULL,
+    city varchar(3) NOT NULL,
+    n_infection varchar(205),
+    city_code varchar(5) NOT NULL,
+    remarks varchar(200)
+);
+ALTER TABLE ONLY infection_numbers_influenza ATTACH PARTITION infection_numbers_influenza_20_01 FOR VALUES IN ('20-01');
```
Comment (kou) [inline sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql:19]: Can we use `CREATE TABLE infection_numbers_influenza_20_01 PARTITION OF infection_numbers_influenza FOR VALUES IN ('20-01')`?

**Review by kou — COMMENTED** [review]
Do we need all `infection_numbers_influenza_XX_YY` tables? Can we reduce them?

**Comment by komainu8** [comment]
> Do we need all `infection_numbers_influenza_XX_YY` tables? Can we reduce them?

I can't reduce  `infection_numbers_influenza_XX_YY` tables.
Because if the number of these tables less than now, PGroonga doesn't crash.

File: sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    remarks varchar(200)
+);
+
+CREATE TABLE infection_numbers_influenza_22_04 PARTITION OF infection_numbers_influenza FOR VALUES IN ('22-04') (
+    country varchar(16) NOT NULL,
+    prefecture varchar(4) NOT NULL,
+    city varchar(3) NOT NULL,
+    n_infection varchar(205),
+    city_code varchar(5) NOT NULL,
+    remarks varchar(200)
```
Comment (kou) [inline sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql:116]: Do we still need these column definitions?

File: sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql (hunk 1 above)
Comment (komainu8) [reply sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql]: I've fixed in https://github.com/pgroonga/pgroonga/pull/236/commits/d7c642bdd3575bcd199031d7b44147b28eeb0296.

File: sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql (hunk 2 above)
Comment (komainu8) [reply sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql]: I've fixed in https://github.com/pgroonga/pgroonga/pull/236/commits/48e6fd67c664f16eb5856c7f01b513278990ffb9.

File: sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql (hunk 3 above)
Comment (komainu8) [reply sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql]: I removed these column definitions in https://github.com/pgroonga/pgroonga/pull/236/commits/c941e20463aea537b51131e2f0036d3f9f6df09c.

**Comment by komainu8** [comment]
@kou Thank you for your comments. I've fixed them.

File: sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+INSERT INTO infection_numbers_influenza_21_04 VALUES ('Germany','BER','BER','221','21-04','age: 50, gender:male, overseas travel history: false');
+INSERT INTO infection_numbers_influenza_22_04 VALUES ('U.S.A','NY','NY','10221','22-04','age: 10, gender:male, overseas travel history: false');
+
+CREATE INDEX remarks_index ON infection_numbers_influenza USING pgroonga (remarks);
+
+SELECT t1.n_infection, t1.remarks
+  FROM infection_numbers_influenza t1
+  INNER JOIN infection_numbers_influenza t2
+    ON t1.city = t2.prefecture
+ WHERE t1.remarks  &@~ 'age';
```
Comment (kou) [inline sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql:34]: Could you also add the following like other tests to confirm query plan too?

```sql
EXPLAIN (COSTS OFF)
SELECT t1.n_infection, t1.remarks
  FROM infection_numbers_influenza t1
  INNER JOIN infection_numbers_influenza t2
    ON t1.city = t2.prefecture
 WHERE t1.remarks  &@~ 'age';
 ```

Comment (komainu8) [reply sql/full-text-search/varchar/query-v2/partitioning/many-partition.sql:44]: I added tests to confirm query plan in https://github.com/pgroonga/pgroonga/pull/236/commits/43c856857fa6daf61f9cfad46741ca49f9198e46.

**Comment by komainu8** [comment]
@kou Thank you for your comment. I've fixed it.

**Pull request merged**